arch/xtensa/core/xtensa_mmu.c

```diff
@@ -32,6 +32,12 @@
 /* Fixed data TLB way to map VECBASE */
 #define MMU_VECBASE_WAY 8
 
+/* Skip TLB IPI when updating page tables.
+ * This allows us to send IPI only after the last
+ * changes of a series.
+ */
```

Comment:
Replying to this patch as a whole:

> When adding a thread to a memory domain, we need to also update
> the mapped page table if it is the current running thread on
> the same CPU. If it's not on the same CPU, we need to notify
> the other CPUs in case the thread is running in one of them.

Wait, why?  Surely we don't asychronously add **running** threads to a domain, no?  That seems crazy to me.  Either you're adding a suspended thread, or you're adding _current.  No way should it ever be legal to try to swap the entire memory map of a thread while it's spinning on another CPU.  No OS does that.   Do we actually have a test that tries this?   Can we just say no and fix the API?